Default.mk

```diff
@@ -0,0 +1,57 @@
+.PHONY: clean obj_dirs .FORCE
+
+.DEFAULT_GOAL := $(TARGET)
+
+.FORCE:
+
+# libProtocols uses non-standard file extensions so it handles it's own objects list
+ifndef OBJECTS
+OBJECTS := $(patsubst $(SRC_DIR)/%, $(OBJ_DIR)/%, $(patsubst %.cpp, %.o, $(SOURCES)))
+endif
+
+OBJ_DIRS = $(sort $(OBJ_DIR) $(dir $(OBJECTS)))
+DEPENDS = $(OBJECTS:.o=.d)
+
+ifdef SHARED_SRC_DIR
+CXXFLAGS += -I$(SHARED_SRC_DIR) $(addprefix -I$(SHARED_SRC_DIR)/, $(SHARED_INCLUDES))
+SOURCES += $(addprefix $(SHARED_SRC_DIR),$(SHARED_SOURCES))
+OBJECTS += $(addprefix $(OBJ_DIR)/shared/,$(SHARED_SOURCES:.cpp=.o))
+DEPENDS += $(addprefix $(OBJ_DIR)/shared/,$(SHARED_SOURCES:.cpp=.d))
+endif
+
+$(TARGET): $(OBJECTS)
```

Comment:
I might suggest moving this into each makefile instead of putting it here, although I see it gives you foo-clean and DEFAULT_GOAL for free.